mitodl/bootcamp-ecommerce
# Ab/add bootcamp foreign key to line

#### What are the relevant tickets?
closes https://github.com/mitodl/bootcamp-ecommerce/issues/893

#### What's this PR do?
Currently Line objects join to BootcampRun objects using the run_key

This pr adds a bootcamp_id foreign key to Line and updates the code to use it to to join to BootcampRun objects.

It also adds a uniqueness constraint on run_key. There are no current duplicate run keys either the production database or on RC so we can do this without changing any of the data.

There will be a follow up pr to remove run_key from  Line.

#### How should this be manually tested?
Verify that you can create charges and refunds and the resulting Line objects have both a bootcamp_id and a run_key

http://bc.odl.local:8099/admin/ecommerce/line/ should have a bootcamp run id

`ecommerce.api.serialize_user_bootcamp_run()` should work as expected

`LineSerializer(Line.objects.last()).data` should return the run_key as expected

`http://localhost:8099/statement/<run id>` should be unchanged
`http://bc.odl.local:8099/applications/<aplication id>/payment-history/` should be unchanged

It's worth testing this PR a bit more thoroughly then usual, because this is my first bootcamps PR
